src/components/App/App.js

```diff
@@ -1,16 +1,22 @@
 import React from 'react'
 import { CssVariables } from '@dhis2/ui'
+import { useLocale } from '../../hooks/locale'
+import { LocaleContext } from '../Context'
 import { Routes } from '../Routes'
 import { PageWrapper } from '../PageWrapper'
 import './App.css'
 
-const App = () => (
-    <React.Fragment>
-        <CssVariables spacers colors />
-        <PageWrapper>
-            <Routes />
-        </PageWrapper>
-    </React.Fragment>
-)
+const App = () => {
+    const locale = useLocale()
+
+    return (
+        <LocaleContext.Provider value={locale}>
+            <CssVariables spacers colors />
+            <PageWrapper>
+                <Routes />
+            </PageWrapper>
+        </LocaleContext.Provider>
+    )
```

Comment:
I'm not sure that unconditionally rendering the app is a good thing here. Effectively this is first going to render the app with an empty locale string first and then re-render the app with the correct locale. IMO it'd be more appropriate to prevent the app from rendering until the local has been set correctly.

Reply:
Yeah, all the components that use locale can cope with an absence of the locale though. And there are just two locations where it's being used. I prefer rendering something while data is being fetched over blocking the ui from rendering while data is being fetched.